### PR TITLE
cmd/snap-bootstrap/initramfs-mounts: write active boot-flags to /run 

### PR DESCRIPTION
**Pull request by anonymouse64**
Based on top of https://github.com/snapcore/snapd/pull/10103, the main commit is 7aef344.

This will enable userspace to inspect unambiguously the active boot flags for
this boot by reading from the file in /run, and writing to the modeenv to
manipulate the next boot flags.

### CODE REVIEW

File: cmd/snap-bootstrap/cmd_initramfs_mounts.go [hunk 1]
```diff
@@ -136,14 +136,48 @@ func generateInitramfsMounts() (err error) {
...
+		// in the case  of install comes from untrusted input, the bootenv. In
+		// the case of run mode, boot flags are read from the modeenv, which
+		// should be valid and trusted, but if the modeenv becomes corrupted, we
+		// would block accessing the system (except through an initramfs shell),
+		// to recover the modeenv (though maybe we could enable some sort of
+		// fixing from recover mode instead?)
+		logger.Noticef("error accessing boot flags: %v", err)
+	} else {
+		// write the boot flags
+		if err := boot.InitramfsSetBootFlags(flags); err != nil {
```
Comment (bboozzoo) [inline cmd/snap-bootstrap/cmd_initramfs_mounts.go:171]: Looking back at #10103, perhaps this should have been named along the lines of `InitramfsExportBootFlagsForSystem`?

File: cmd/snap-bootstrap/cmd_initramfs_mounts.go [hunk 2]
```diff
@@ -136,14 +136,48 @@ func generateInitramfsMounts() (err error) {
...
+			// TODO: should we die here if we can't write the boot flags?
+			//       boot flags are written to /run, so if we can't write there
+			//       arguably we would have other major problems so it's
+			//       something of a moot point
```
Comment (pedronis) [inline cmd/snap-bootstrap/cmd_initramfs_mounts.go:175]: I agree,  we should probably convert this TODO into:

// cannot write to /run, error

File: cmd/snap-bootstrap/cmd_initramfs_mounts.go (hunk 1 above)
Comment (pedronis) [reply cmd/snap-bootstrap/cmd_initramfs_mounts.go:171]: maybe Expose or Enable ...ForSystem ?

**Review by pedronis — APPROVED** [review]
thank you, small comments

File: cmd/snap-bootstrap/cmd_initramfs_mounts.go (hunk 1 above)
Comment (anonymouse64) [reply cmd/snap-bootstrap/cmd_initramfs_mounts.go:171]: I like InitramfsExposeBootFlagsForSystem

File: cmd/snap-bootstrap/cmd_initramfs_mounts.go (hunk 2 above)
Comment (anonymouse64) [reply cmd/snap-bootstrap/cmd_initramfs_mounts.go]: fixed this up

**Review by bboozzoo — APPROVED** [review]
LGTM

**Pull request merged**